src/olympia/hero/tests/test_models.py

```diff
@@ -231,6 +232,27 @@ def test_clean_cta_remove_prefixes(self):
             hero.clean()
             assert hero.cta_url == '/en-US/firefox/something/weird/'
 
+    def test_clean_cta_remove_prefixes_with_site_url(self):
+        hero = SecondaryHero.objects.create()
+
+        with self.activate(locale='en-US', app='firefox'):
+            hero.cta_url = 'http://testserver/en-US/firefox/addon/foo'
+            hero.clean()
+            assert hero.cta_url == '/addon/foo/'
+
+        with self.activate(locale='en-US', app='firefox'):
+            # With default test settings, this isn't transformed because
+            # neither the SITE_URL nor the EXTERNAL_SITE_URL match.
+            hero.cta_url = 'https://addons.mozilla.org/en-US/firefox/addon/foo'
+            hero.clean()
+            assert hero.cta_url == 'https://addons.mozilla.org/en-US/firefox/addon/foo'
+
+            # When overriding the EXTERNAL_SITE_URL however, it works.
+            with override_settings(EXTERNAL_SITE_URL='https://addons.mozilla.org'):
+                hero.cta_url = 'http://testserver/en-US/firefox/addon/foo'
+                hero.clean()
+                assert hero.cta_url == '/addon/foo/'
+
```

Comment:
Also, can we repeat with `SITE_URL`?

Reply:
The initial assertion at the top of this method is doing that, since `SITE_URL is `http://testserver` in tests.